ext/statistics/main.php

```diff
@@ -0,0 +1,147 @@
+                arsort($tag_tally, $flags = SORT_NUMERIC);
+                $tag_table = $this->theme->build_table($tag_tally, "Taggers", "Top $limit taggers", $limit);
+            } else {
+                $tag_table = null;
+            }
+
+            $upload_tally = [];
+            foreach ($this->get_upload_stats() as $name) {
+                array_key_exists($name, $upload_tally) ? $upload_tally[$name] += 1 : $upload_tally[$name] = 1;
+
+            }
+            arsort($upload_tally, $flags = SORT_NUMERIC);
+            $upload_table = $this->theme->build_table($upload_tally, "Uploaders", "Top $limit uploaders", $limit);
+
+            if (Extension::is_enabled(CommentListInfo::KEY)) {
+                $comment_tally = [];
+                foreach ($this->get_comment_stats() as $name) {
+                    array_key_exists($name, $comment_tally) ? $comment_tally[$name] += 1 : $comment_tally[$name] = 1;
+
+                }
+                arsort($comment_tally, $flags = SORT_NUMERIC);
+                $comment_table = $this->theme->build_table($comment_tally, "Commenters", "Top $limit commenters", $limit);
+            } else {
+                $comment_table = null;
+            }
+
+            if (Extension::is_enabled(FavoritesInfo::KEY)) {
+                $favorite_tally = [];
+                foreach ($this->get_favorite_stats() as $name) {
+                    array_key_exists($name, $favorite_tally) ? $favorite_tally[$name] += 1 : $favorite_tally[$name] = 1;
+
+                }
+                arsort($favorite_tally, $flags = SORT_NUMERIC);
+                $favorite_table = $this->theme->build_table($favorite_tally, "Favoriters", "Top $limit favoriters", $limit);
+            } else {
+                $favorite_table = null;
+            }
+
+            $this->theme->display_page($page, $limit, $tag_table, $upload_table, $comment_table, $favorite_table);
+        }
+    }
+
+    public function onSetupBuilding(SetupBuildingEvent $event): void
+    {
+        $sb = $event->panel->create_new_block("Stats Page");
+        $sb->add_longtext_option("stats_text", "<br>Page Text:<br>");
+    }
+
+    public function onPageNavBuilding(PageNavBuildingEvent $event): void
+    {
+        $event->add_nav_link("stats", new Link('stats'), "Stats");
+    }
+
+    public function onPageSubNavBuilding(PageSubNavBuildingEvent $event): void
+    {
+        if ($event->parent == "stats") {
+            $event->add_nav_link("stats_100", new Link('stats/100'), "Top 100");
+        }
+    }
+
+    /**
+     * @return array<string, int>
+     */
+    private function get_tag_stats(): array
+    {
+        global $database;
+        // Returns the username and tags from each tag history entry. Excludes Anonymous
+        $tag_stats = $database->get_all('SELECT users.name,tag_histories.tags,tag_histories.image_id FROM tag_histories INNER JOIN users ON users.id = tag_histories.user_id WHERE tag_histories.user_id <> 1;');
```

Comment:
In some rare cases anon's ID can be something other than 1 -- `$config->get_int("anon_id")` should always get the correct ID